lucene/core/src/java/org/apache/lucene/index/IndexingChain.java

```diff
@@ -349,7 +350,8 @@ private void writePoints(SegmentWriteState state, Sorter.DocMap sortMap) throws
 
             perField.pointValuesWriter.flush(state, sortMap, pointsWriter);
             perField.pointValuesWriter = null;
-          } else if (perField.fieldInfo.getPointDimensionCount() != 0) {
+          } else if (perField.fieldInfo != null
```

Comment:
shouldn't fieldInfo always be non null when reaching here?

Reply:
Not, always.
We first create PerField: `PerField pf = getOrAddPerField(field.name(), fieldType);` and then initialize its FieldInfo `initializeFieldInfo(pf)`. Initialization of FieldInfo may fail, if there is already a global field with the same name but a different schema. So we may end up with `PerField` with a null `fieldInfo`.